packages/video_player/video_player/lib/video_player.dart

```diff
@@ -397,9 +407,8 @@ class VideoPlayerController extends ValueNotifier<VideoPlayerValue> {
       }
     }
 
-    if (closedCaptionFile != null) {
-      _closedCaptionFile ??= await closedCaptionFile;
-      value = value.copyWith(caption: _getCaptionAt(value.position));
+    if (_closedCaptionFileFuture != null) {
+      await setClosedCaptionFile(_closedCaptionFileFuture);
```

Comment:
Calling the setter to set the same value that the field is already set to in order to benefit from side-effects in the setter is an anti-pattern; it makes the code less understandable, and is subject to subtle breakage if someone later optimizes the setter to be a no-op if the new and old values are the same.

The way to share code here would be to make a new `_updateClosedCaptionWithFuture(...)` method and put the logic there, then implement both `initialize` and `setClosedCaptionFile` using that.

Reply:
I guess everything is okay now.